test/torchaudio_unittest/backend/soundfile/common.py

```diff
@@ -32,3 +32,26 @@ def skipIfFormatNotSupported(fmt):
 
 def parameterize(*params):
     return parameterized.expand(list(itertools.product(*params)), name_func=name_func)
+
+
+def fetch_wav_subtype(dtype, encoding, bits_per_sample):
+    subtype = {
+        (None, None): dtype2subtype(dtype),
+        (None, 8): "PCM_U8",
+        ('PCM_U', None): "PCM_U8",
+        ('PCM_U', 8): "PCM_U8",
+        ('PCM_S', None): "PCM_32",
+        ('PCM_S', 16): "PCM_16",
+        ('PCM_S', 32): "PCM_32",
+        ('PCM_F', None): "FLOAT",
+        ('PCM_F', 32): "FLOAT",
+        ('PCM_F', 64): "DOUBLE",
+        ('ULAW', None): "ULAW",
+        ('ULAW', 8): "ULAW",
+        ('ALAW', None): "ALAW",
+        ('ALAW', 8): "ALAW",
+    }.get((encoding, bits_per_sample))
+    if subtype:
+        return subtype
```

Comment:
I recommend checking the key first as using `get` method on dictionary, you might get `None` value for an existing key.

```python
key = (encoding, bits_per_sample)
if key not in subtypes:
    raise ...
return subtypes[key]
```

Reply:
We would get `None` for keys that are not present, in which case we raise ValueError(line 56). Why do we need `if`?

Reply:
Sorry, my comment is not making sense. I wanted to say, in general a key missing and a key mapped to `None` value are semantically different. Checking "if the value returned by `get` method on dictionary" is checking if either "key is missing" **OR** "key is mapped to `None`". This slightly feels off (maybe only to me) because the purpose of this helper function is to define the exact mapping that will be used by the tests, so checking if the key exists is the right thing to do and that's the right amount of checking. I feel that checking key explicitly better expresses the intent of the code. Of course it is no harm to check the `None` here, but it makes me wonder "is that the responsibility of the helper function?".

Okay, my points got very subjective, and now I feel doing more check (against None) feels better. so let's keep it as is.